src/silverlabnwb/nwb_file.py

```diff
@@ -798,12 +810,17 @@ def _write_roi_data(self, all_rois, num_trials, cycles_per_trial,
                     # TODO: Consider precision: the round() here is to match the exported data...
                     ch_data = np.round(tdms_file.channel_data('Functional Imaging Data',
                                                               'Channel {} Data'.format(ch)))
-                    ch_data = ch_data.reshape(ch_data_shape)
                     # Copy each ROI's data into the NWB
+                    offset_from_previous_rois = 0
                     for roi_num, data_paths in all_rois.items():
+                        roi_shape = all_roi_dimensions_pixels[roi_num]
+                        roi_ch_data = ch_data[offset_from_previous_rois:
+                                              offset_from_previous_rois + cycles_per_trial*roi_shape[0]*roi_shape[1]]
+                        roi_ch_data = roi_ch_data.reshape(np.concatenate((roi_shape, [cycles_per_trial]))[::-1])
```

Comment:
Right, as you say, this isn't the right chunk of data because of how the TDMS file is laid out. We can get the correct indices, though!